### ISSUE TITLE
local type Type = module.Type results in stack overflow

### ISSUE THREAD

**Issue by Grinshpon** [teal-language/tl#238]
I've made a small example showcasing the issue:

`module.tl`:

```
local record module
  record Type
    data: number
  end
end
return module
```

`main.tl`:

```lua
local module = require "module"
local type Type = module.Type
local var: Type = {data = 0}
print(var.data)
```

When I try to compile the scripts, there is a stack overflow:

```
/usr/local/bin/lua: /usr/local/share/lua/5.3/tl.lua:4568: stack overflow
stack traceback:
	/usr/local/share/lua/5.3/tl.lua:4568: in upvalue 'is_a'
	/usr/local/share/lua/5.3/tl.lua:4651: in upvalue 'is_a'
	/usr/local/share/lua/5.3/tl.lua:4651: in upvalue 'is_a'
	/usr/local/share/lua/5.3/tl.lua:4651: in upvalue 'is_a'
	/usr/local/share/lua/5.3/tl.lua:4651: in upvalue 'is_a'
	/usr/local/share/lua/5.3/tl.lua:4651: in upvalue 'is_a'
	/usr/local/share/lua/5.3/tl.lua:4651: in upvalue 'is_a'
	/usr/local/share/lua/5.3/tl.lua:4651: in upvalue 'is_a'
	/usr/local/share/lua/5.3/tl.lua:4651: in upvalue 'is_a'
	/usr/local/share/lua/5.3/tl.lua:4651: in upvalue 'is_a'
	...
	/usr/local/share/lua/5.3/tl.lua:4651: in upvalue 'is_a'
	/usr/local/share/lua/5.3/tl.lua:4651: in upvalue 'is_a'
	/usr/local/share/lua/5.3/tl.lua:4778: in upvalue 'assert_is_a'
	/usr/local/share/lua/5.3/tl.lua:5797: in field 'after'
	/usr/local/share/lua/5.3/tl.lua:2238: in function </usr/local/share/lua/5.3/tl.lua:2232>
	(...tail calls...)
	/usr/local/share/lua/5.3/tl.lua:2327: in upvalue 'recurse_node'
	/usr/local/share/lua/5.3/tl.lua:6617: in function 'tl.type_check'
	/usr/local/share/lua/5.3/tl.lua:6723: in function 'tl.process_string'
	/usr/local/share/lua/5.3/tl.lua:6668: in function 'tl.process'
	/usr/local/lib/luarocks/rocks-5.3/tl/0.8.0-1/bin/tl:760: in main chunk
	[C]: in ?

```

However, when I modify `main.lua` to omit the type definition and use `module.Type` directly, it compiles just fine.

**Comment by Grinshpon** [comment]
Oddly enough, if I keep the type definition in `main.lua` but don't use it, it compiles as well:

```
local module = require "module"
local type Type = module.Type
local var: module.Type = {data = 0}
print(var.data)
```

**Comment by pdesaulniers** [comment]
Also, if we change the line to
```lua
local Type = module.Type
```
Then, it compiles as expected.

**Comment by hishamhm** [comment]
@Grinshpon Thank you for the report! Will look into it!

**Comment by hishamhm** [comment]
Should be fixed now with 6575c6d!
